### PR TITLE
Ref socketio after ref engineio

### PR DESCRIPTION
**Pull request by mrfoe7**
Need review after PR https://github.com/googollee/go-socket.io/pull/455

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -25,74 +25,30 @@ type Conn interface {
...
-	n.mu.RLock()
-	defer n.mu.RUnlock()
-	for ns, nc := range n.namespaces {
-		fn(ns, nc)
-	}
-}
-
 type conn struct {
+	*parser.Encoder
+	*parser.Decoder
```
Comment (googollee) [inline connection.go:30]: It exposes Encode/Decode to users. 

I know it's convenient to write `conn.Encode()` than `conn.encoder.Encode()`, but I hope exposing methods could give a guide that which one could be called by caller, and which one should not. Exposing Encode/Decode breaks that rule.

Comment (mrfoe7) [reply connection.go]: > It exposes Encode/Decode to users.

Sure, sorry, my fault. Rollbacked

File: connection.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+			return
+		}
+
+		if header.Namespace == aliasRootNamespace {
+			header.Namespace = rootNamespace
+		}
+
+		var err error
+		switch header.Type{
+		case parser.Ack, parser.Connect, parser.Disconnect:
```
Comment (mrfoe7) [inline connection.go:215]: why we not handle `parser.Error`?

Comment (googollee) [reply connection.go:215]: You are right, it should be here too.

Comment (mrfoe7) [reply connection.go:215]: Issue https://github.com/googollee/go-socket.io/issues/466

**Pull request merged**